lib/node/createIndex.ts

```diff
@@ -3,23 +3,24 @@
 import sliced from 'sliced';
 import { Node } from '.';
 import { Column } from '../column';
-import { Table } from '../table';
+import { IndexCreationQuery, Table } from '../table';
 
-export class CreateIndexNode extends Node {
+export class CreateIndexNode extends Node implements IndexCreationQuery {
     public table: Table<unknown>;
     public options: {
         indexName: string;
         columns: Column<unknown>[];
         type?: string;
         algorithm?: string;
         parser?: string;
+        ifNotExists: boolean;
     };
 
     constructor(table: Table<unknown>, indexName: string) {
         super('CREATE INDEX');
 
         this.table = table;
-        this.options = { indexName, columns: [] };
+        this.options = { indexName, columns: [], ifNotExists: false };
```

Comment:
please remove default false value as this new property will only have effect when true